src/data/adHocFilter.ts

```diff
@@ -26,23 +26,33 @@ export class AdHocFilter {
     if (this._targetTable === '' || !sql.match(new RegExp(`.*\\b${this._targetTable}\\b.*`, 'gi'))) {
       return sql;
     }
+
     let filters = adHocFilters
+      .filter(validate)
       .map((f, i) => {
         const key = f.key.includes('.') ? f.key.split('.')[1] : f.key;
         const value = isNaN(Number(f.value)) ? `\\'${f.value}\\'` : Number(f.value);
         const condition = i !== adHocFilters.length - 1 ? (f.condition ? f.condition : 'AND') : '';
         return ` ${key} ${f.operator} ${value} ${condition}`;
       })
       .join('');
+
+    if(filters === '') {
+      return sql;
+    }
     // Semicolons are not required and cause problems when building the SQL
     sql = sql.replace(';', '');
     return `${sql} settings additional_table_filters={'${this._targetTable}' : '${filters}'}`;
   }
 }
 
+function validate(filter: AdHocVariableFilter): boolean {
+  return filter.key !== undefined && filter.operator !== undefined && filter.value !== undefined;
```

Comment:
I didn't add a test for `filter.key: undefined` since here we assume it's a string https://github.com/grafana/clickhouse-datasource/pull/451/files#diff-432fcb747320944461075f7baa45beacdd3b15b65cf530690f35030d978c5e3dR23
Should I add at the very top this condition?
```js
if(filter.key === undefined) return sql
```

Reply:
I like this idea. We should probably have had that to begin with given the `includes` line you linked to.